src/guide/keys.jl

```diff
@@ -166,16 +164,96 @@ function render_discrete_key(labels::Vector{String}, title_ctx::Context, title_w
                             svgclass("guide colorkey"))
         end
 
-        return compose!(
-            context(minwidth=max(title_width, ctxwidth),
-                    minheight=ctxheight,
-                    units=UnitBox()),
-            ctxp)
+        return compose!(context(minwidth=max(title_width, ctxwidth),
+                    minheight=ctxheight, units=UnitBox()),
+                    ctxp)
     end
 
     return map(make_layout, 1:maxcols)
 end
 
 
+function render_key_title2(title::AbstractString, theme::Gadfly.Theme)
+    title_width, title_height = max_text_extents(theme.key_title_font, theme.key_title_font_size, title)
+
+    if theme.guide_title_position == :left
+        title_form = text(0.0w, title_height, title, hleft, vbottom)
+    elseif theme.guide_title_position == :center
+        title_form = text(0.5w, title_height, title, hcenter, vbottom)
+    elseif theme.guide_title_position == :right
+        title_form = text(1.0w, title_height, title, hright, vbottom)
+    else
+        error("$(theme.guide_title_position) is not a valid guide title position")
+    end
+
+    title_padding = 1.5mm
```

Comment:
the only difference i can see between `render_key_title` and `render_key_title2` is the use of `4mm` in the former here.  to eliminate such code duplication, would it be possible to pass in `title_padding` as an argument?

Reply:
I would prefer to keep the two functions separate for the moment, as I'm working towards separating the keys for discrete and continuous aesthetics.  After I complete that I'm happy to tidy up whatever's remaining.